clientv3/watch.go

```diff
@@ -61,6 +61,11 @@ func (wr *WatchResponse) Err() error {
 	return nil
 }
 
+// IsProgressNotify returns true if the WatchResponse is progress notification.
+func (wr *WatchResponse) IsProgressNotify() bool {
+	return len(wr.Events) == 0 && !wr.Canceled
```

Comment:
Client initiated channel cancels don't deliver `Canceled` since the client already knows the stream is closing. Other than that, you're right. `Canceled` is mostly there to know whether the watcher lost the stream / couldn't establish the stream because of compaction.